sherpa/__init__.py

```diff
@@ -116,11 +116,25 @@ def test(level=1, verbosity=1, datadir=None):
 
 
 def clitest():
-    try:
-        import pytest, pytest_cov
-        errno = pytest.main(['--pyargs', 'sherpa', '-rs'])
-        sys.exit(errno)
-    except ImportError:
-        print """Cannot import pytest and pytest-cov.
-            Please run 'pip install -r test_requirements.txt' first"""
-        sys.exit(1)
+    def install(package_name):
+        try:
+            import pip
+            pip.main(['install', package_name])
+        except:
+            print """Cannot import pip or install packages with it.
+            You need pytest, and possibly pytest-cov, in order to run the tests.
+            If you downloaded the source code, please run 'pip install -r test_requirements.txt' first.
+            """
```

Comment:
That's a good idea.

clitest is bound to the sherpa_test console script in setup.py... it is called when sherpa_test is called, so there is no guarantee it's called from the right folder.